src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,221 @@
+                + "FROM cars AS c "
+                + "JOIN manufacturers AS m ON c.manufacturer_id = m.id "
+                + "WHERE c.id = ? AND c.is_deleted = false;";
+        Car car = null;
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                car = parseCarFromResultSet(resultSet);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't get driver by id " + id, e);
+        }
+        if (car != null) {
+            car.setDrivers(getDriversForCar(id));
+        }
+        return car;
+    }
+
+    @Override
+    public List<Car> getAll() {
+        String query = "SELECT c.id AS car_id, model, manufacturer_id, name, country "
+                + "FROM cars AS c "
+                + "JOIN manufacturers AS m ON c.manufacturer_id = m.id "
+                + "WHERE c.is_deleted = false;";
+        Car car = null;
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                car = parseCarFromResultSet(resultSet);
+                cars.add(car);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't get all cars from carsDB", e);
+        }
+        for (Car automobile : cars) {
+            automobile.setDrivers(getDriversForCar(automobile.getId()));
+        }
+        logger.info("The all data from DB was successful fetched");
+        return cars;
+    }
+
+    @Override
+    public Car update(Car car) {
+        logger.info("Method update was called with car: " + car);
+        String query = "UPDATE cars SET model = ?, manufacturer_id = ? "
+                + "WHERE id = ? AND is_deleted = false;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setString(1, car.getModel());
+            statement.setLong(2, car.getManufacturer().getId());
+            statement.setLong(3, car.getId());
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't update car " + car, e);
+        }
+        deleteDriversFromCar(car);
+        addDriversToCar(car);
+        logger.info("Update data of car was successful");
+        return car;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        logger.info("Method delete was called with id: " + id);
+        String query = " UPDATE cars SET is_deleted = TRUE WHERE id = ?;";
```

Comment:
(suggestion only) There's an reduntant space at the beginning of query, but it's not critically)